sos/plugins/openstack_ironic.py

```diff
@@ -122,8 +125,18 @@ class RedHatIronic(OpenStackIronic, RedHatPlugin):
         'openstack-ironic-discoverd-ramdisk'
     ]
 
+    def collect_introspection_data(self):
+        uuids_result = self.call_ext_prog('openstack baremetal node list '
+                                          '-f value -c UUID')
+        uuids = [uuid for uuid in uuids_result['output'].split()
```

Comment:
What if the `openstack baremetal node list` command terminates with an error? Isnt it worth checking `if uuids_result["status"] == 0` first?

Reply:
As this command is called in the parent class, I'm not sure how sosreport will behave.. Anyway, this is the most basic ironic command, if it fails, it's fine to fail the whole collector.

Reply:
That does not excuse you from having to perform normal validation: exceptions in plugin classes (which is the likely result of continuing to try to operate without checking the command status...) are never acceptable.

Also, if this is collected already in the base class, why are we running it _again_ here? Why wasn't the superclass changed to store the result somewhere for further processing?

Reply:
> Why wasn't the superclass changed to store the result somewhere for further processing?

Probably because I'm not a sosreport guru :) I think I get it, lemme try.

Reply:
Aha, now I remember. Because we collect nodes in the base class as a table with all fields, and here we only collect UUIDs. I don't think we should parse tables here, so I guess I'll go with "normal validation" option then.